src/db.rs

```diff
@@ -245,15 +273,22 @@ impl Database {
     /// * if the file does not exist, or is an empty file, a new database will be initialized in it
     /// * if the file is a valid redb database, it will be opened
     /// * otherwise this function will return an error
+    #[cfg(any(windows, unix, target_os = "wasi"))]
     pub fn create(path: impl AsRef<Path>) -> Result<Database, DatabaseError> {
         Self::builder().create(path)
     }
 
     /// Opens an existing redb database.
+    #[cfg(any(windows, unix, target_os = "wasi"))]
```

Comment:
I'm especially unexcited about making the public API platform dependent. Why is this necessary?

Reply:
Thank you for reviewing my changes. The reason is that the filesystem API is unavailable on certain targets (for instance `wasm32-unknown-unknown`) so without platform-gating these methods the crate fails to compile. This happens even on `master` currently, since an implementation for `LockedFile` is missing.

The two options that I see for working around this are:
- Making certain public methods only available on certain platforms (what I currently have)
- Creating a dummy `FileBackend` implementation for non-supported targets that just panics immediately at runtime

I initially went with the first option, but if you think the second one is preferable (or have a separate solution) please let me know and I will change it.

Reply:
Ah, I see. `wasm64-unknown-unknown` is Tier 3, so I'd consider any support of it experimental at best. I'd like to keep the codebase as free of `cfg` as possible, because it adds maintenance. So ya, lets remove these and if it's possible to cleanly implement option 2 for that wasm target, I'm fine having a separate backend for it.

Alternately, if there's a flag that indicates the filesystem API is available on a platform, then I'd be fine using that here. Basically, I want this to work on all platform where it's supported, without the need to add those platforms explicitly to this `cfg`